File: Server/scrumer/src/main/java/com/ringff/scrumer/common/RFHttpResponse.java
package com.ringff.scrumer.common;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

/**
 * Ringff http response
 * @author Eric Zhao 
 *
 * @param <T>
 */
@JsonPropertyOrder({RFHttpResponse.Field_header, BaseResponse.Field_data})
public class RFHttpResponse<T> extends BaseResponse<T> {

    public static final String Field_header = "header";
    
    private ResponseHeader header;
    
    private RFHttpResponse(){
        this.header = new ResponseHeader();
    }
    
    public static <T>  RFHttpResponse<T> build(){
        return new RFHttpResponse<>();
    }
    
    public RFHttpResponse<T> setStatus(EnumResponseStatus status){
        this.header.setStatus(status);
        return this;
    }
    public RFHttpResponse<T> setMessage(String message){
        this.header.setMessage(message);
        return this;
    }
    
    @JsonProperty(RFHttpResponse.Field_header)
    public ResponseHeader getHeader(){
        return this.header;
    }
    
}

File: Client/Ringff.Common.Object/Common/RFHttpResponse.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;


/************************************************************************   
 * Version :  v1.0.0.0  
 * Description :      
 * Author :  Eric Zhao 
************************************************************************/
namespace Ringff.Common.Object.Common
{
    public class RFHttpResponse<T> : BaseResponse<T>
    {
        public RFHttpResponse()
        {
            Header = new ResponseHeader();
        }

        public const String Field_header = "header";

        [JsonProperty(PropertyName = "header",Order=1)]
        public ResponseHeader Header
        {
            get;
            set;
        }
    }
}